specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2023-09-01/Workspaces.json

```diff
@@ -0,0 +1,803 @@
+        },
+        "createdDate": {
+          "type": "string",
+          "description": "Workspace creation date.",
+          "readOnly": true
+        },
+        "modifiedDate": {
+          "type": "string",
+          "description": "Workspace modification date.",
+          "readOnly": true
+        },
+        "publicNetworkAccessForIngestion": {
+          "description": "The network access type for accessing Log Analytics ingestion.",
+          "$ref": "#/definitions/PublicNetworkAccessType"
+        },
+        "publicNetworkAccessForQuery": {
+          "description": "The network access type for accessing Log Analytics query.",
+          "$ref": "#/definitions/PublicNetworkAccessType"
+        },
+        "forceCmkForQuery": {
+          "type": "boolean",
+          "description": "Indicates whether customer managed storage is mandatory for query management."
+        },
+        "privateLinkScopedResources": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkScopedResource"
+          },
+          "x-ms-identifiers": [
+            "resourceId"
+          ],
+          "description": "List of linked private link scope resources."
+        },
+        "features": {
+          "$ref": "#/definitions/WorkspaceFeatures",
+          "description": "Workspace features."
+        },
+        "defaultDataCollectionRuleResourceId": {
+          "type": "string",
+          "description": "The resource ID of the default Data Collection Rule to use for this workspace. Expected format is - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/dataCollectionRules/{dcrName}."
+        }
+      },
+      "description": "Workspace properties."
+    },
+    "WorkspaceFeatures": {
+      "type": "object",
+      "properties": {
+        "enableDataExport": {
+          "type": "boolean",
+          "x-nullable": true,
+          "description": "Flag that indicate if data should be exported."
+        },
+        "immediatePurgeDataOn30Days": {
+          "type": "boolean",
+          "x-nullable": true,
+          "description": "Flag that describes if we want to remove the data after 30 days."
+        },
+        "enableLogAccessUsingOnlyResourcePermissions": {
+          "type": "boolean",
+          "x-nullable": true,
+          "description": "Flag that indicate which permission to use - resource or workspace or both."
+        },
+        "clusterResourceId": {
+          "type": "string",
+          "x-nullable": true,
+          "description": "Dedicated LA cluster resourceId that is linked to the workspaces."
+        },
+        "disableLocalAuth": {
+          "type": "boolean",
+          "x-nullable": true,
+          "description": "Disable Non-AAD based Auth."
+        },
+        "unifiedSentinelBillingOnly": {
+          "type": "boolean",
+          "x-nullable": true,
```

Comment:
>     "x-nullable": true, [](http://example.com/codeflow?start=5&length=24)

I would like to discourage you from using nullable booleans, how about saying its optional, or always has a default value, instead?

Reply:
Default value is null, why x-nullable is discouraged? it's an accurate depiction of the property behavior.